underfs/s3/src/main/java/tachyon/underfs/s3/S3UnderFileSystem.java

```diff
@@ -347,6 +333,23 @@ private boolean deleteInternal(String key) {
   }
 
   /**
+   * Gets the child name based on the parent name. If the parent is not valid, the entire child
```

Comment:
I have similar concerns on stripPrefixIfPresent as I think if the cases when s3 prefix doesn't present, it is an error case. 

Though, stripFolderSuffixIfPresent seems totally legitimate to me, as we may see folder or filer which has no suffix.